build-scripts/formatter-emscripten/shell.html

```diff
@@ -0,0 +1,155 @@
+            padding: 16px;
+            display: none;
+        }
+    </style>
+</head>
+
+<body>
+    <div class="row">
+        <div class="col">
+            <label for="json_input">Paste unformatted JSON on left side</label>
+            <textarea id="json_input" placeholder="Paste unformatted JSON on left side"></textarea>
+            <span id="json_error_alert">JSON has errors, can't format!</span>
+        </div>
+        <div class="col">
+            <label for="json_output">Copy formatted JSON from right side</label>
+            <textarea id="json_output" readonly onClick="this.select()"></textarea>
+        </div>
+    </div>
+    {{{ SCRIPT }}}
+    <script>
+
+        let json_input = document.getElementById("json_input");
+        let json_output = document.getElementById("json_output");
+        let json_error_alert = document.getElementById("json_error_alert");
+
+        json_input.placeholder =
+            "\n    Paste your unformatted json into this area, and" +
+            "\n    it will be formatted on the right side." +
+            "\n" +
+            "\n    If there is an error you need to fix it first.";
+
+        json_output.placeholder = "\n    Formatter is loading." +
+            "\n    This tool needs javascript enabled, " +
+            "\n    and a modern browser.";
+
+        json_formatter().then(function (module) {
+            let json_format = module.cwrap("json_format", "string", ["string"]);
+
+            json_output.placeholder =
+                "\n\n    Formatter ready."
+                + "\n    Paste into panel on the left to format.";
+
+            json_input.oninput = function () {
+                if (json_input.value.length < 1) {
+                    json_error_alert.style.display = "none";
+                    json_output.value = "";
+                    return;
+                }
+                // not all invalid json is rejected by the cata formatter,
+                // filter early using the browser's parser instead
+                try {
+                    // https://stackoverflow.com/a/20392392
+                    let o = JSON.parse(json_input.value);
+                    if (o && typeof o === "object") {
+                        let formatted = json_format(json_input.value);
+                        json_error_alert.style.display =
+                            (formatted.startsWith("Json error")
+                                ? "block" : "none");
+                        json_output.value = formatted;
+                        return;
+                    }
+                } catch (e) {
+                    json_error_alert.style.display = "block";
+                    json_output.value = e.message;
+                    return;
+                }
+                json_error_alert.style.display = "block";
+                json_output.value =
+                    "Cataclysm JSON must start with array or object";
+            };
+        });
+
+    </script>
+</body>
+
+</html>
```

Comment:
```suggestion
</html>
```

Reply:
Seems vscode default html formatter trims it regardless of .editorconfig
I added it manually as github ui doesn't let applying [whitespace changes](https://cdn.discordapp.com/attachments/861650119899873321/1122115925706031145/image.png)

I had to move files anyway - changes in build-scripts seem to trigger a full clang-tidy check